cmd/util.go

```diff
@@ -0,0 +1,26 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"os"
+)
+
+// getComponent returns the component to be used for the operation. If an input
+// component is specified, then it is returned, if not, the current component
+// is fetched and returned
+func getComponent(client *occlient.Client, inputComponent, applicationName, projectName string) string {
+	if len(inputComponent) == 0 {
+		c, err := component.GetCurrent(client, applicationName, projectName)
+		checkError(err, "Could not get current component")
+		return c
+	}
+	exists, err := component.Exists(client, inputComponent, applicationName, projectName)
```

Comment:
Do we need to check that component exists if we got it from `GetCurrent`?
`Exists` is expensive it might be enough to put into `else` block for previous if.

Reply:
we're not checking if we're getting it from GetCurrent. The if block has a return statement, we don't need an else here

Reply:
ah, sorry, Didn't notice return there. LGTM than ;-)